test/e2e/marketplace_test.go

```diff
@@ -0,0 +1,186 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	apis "github.com/operator-framework/operator-marketplace/pkg/apis"
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	retryInterval        = time.Second * 5
+	timeout              = time.Second * 60
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+	testNamespace        = "operator-test"
+)
+
+func TestMarketplace(t *testing.T) {
+	operatorsourceList := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+			APIVersion: fmt.Sprintf("%s/%s",
+				operator.SchemeGroupVersion.Group, operator.SchemeGroupVersion.Version),
+		},
+	}
+	catalogsourceconfigList := &operator.CatalogSourceConfig{
```

Comment:
I suggest using the builders for OperatorSource and CatalogSource here. Or are you worried about using them because they don't have unit tests? Comment applies to all places where you are creating these CRs.

Reply:
Yes, I think that adding an additional import (since right now we are only importing the api package which was created by the framework) to use the builders is extra code that could itself introduce unexpected bugs. I do think if they were well tested ahead of time (with something like unit tests) then they would be safer to use.